### DISCUSSION TITLE
Help, copy all settings

### DISCUSSION THREAD

**Discussion by Junglaverde** [pieroproietti/penguins-blog#8] [Q&A] [answered]
Hello, I have tried with sudo eggs tools skel to copy the user configuration. But only the ./config folder is copied.
How can I get the ./local folder to be copied too?
I have copied it directly to just copy the first one.
Thank you very much and congratulations for the work. It's excellent.

**Comment by pieroproietti** [comment]
Hi @Junglaverde, welcome!

Add things to skel, lead to caos... so I tend - as much as possible - to less is better!

I imagine  `.local` as something to not be copied - perhpas I'm wrong.

Anyway if You know to need `.local` to complete your configuration, you can add it manually:  `sudo cp .local /etc/skel/ -R`.

**Reply by Junglaverde** [reply]
Hello, thanks for the answer.
I already copied it but it didn't work. I even copied it to the skel inside eggs before creating the iso. In the live boot it has worked but when I do the installation it has disappeared.
I've thought about copying .local into .config and then making a script at startup that moves .local into place.

**Reply by Junglaverde** [reply]
I have also tried cloning my system. But when I install it, it doesn't create the users and it doesn't let me login.

**Reply by pieroproietti** [reply, marked as answer]
Can't answer, I think (It's my opinion) who ~/.local must not be copied. 

Inside user home there is so big mess and it's difficult to separe things necessary as skeleton and not. Explecially thinkig we can have a lot of differents desktop manager.

All the code of skel is in [xdg.ts](https://github.com/pieroproietti/penguins-eggs/blob/master/src/classes/xdg.ts) you - or any other interested on the project - can try to get a look and suggest something.

**Reply by Junglaverde** [reply]
Okay. She asked if she had an easy solution, and she didn't waste time doing tests.
I will look for how to solve it or pass the configuration to the system directly without using ./local.
Thanks for the attention.
And what was said. Penguins eggs is an excellent project.

**Comment by pieroproietti** [comment]
Another things to Say: when you use --clone the users remain the same as
the parent system. So, probably you just need to login on your cloned
system with your user and password.

Thanks for the comments

Il lun 31 lug 2023, 08:36 Junglaverde ***@***.***> ha scritto:

**Reply by Junglaverde** [reply]
Hello again, thanks for the help. 
In live mode I was able to enter with my username and password. But when I installed it it didn't ask me to set the user during the process and after reboot it wouldn't let me log in with my username and password.
In the end, the idea of ​​creating the script has worked for me. Now I'm going to try changing my ./local settings to the system.
Many thanks for everything.

**Comment by pieroproietti** [comment]
Ok, It happen to be wrong! :-)

As soon a I can will check more... The intended behaviour of --clone is to
let to log to live and to installed system with the old users.


Il lun 31 lug 2023, 10:47 Junglaverde ***@***.***> ha scritto:

**Reply by Junglaverde** [reply]
I am using devuan chimaera in case you need more information.
I have also seen a problem with synaptic after installation. The error was giving /var/lib/apt/lists/lock – open
The folder lists had disappeared.

**Comment by Junglaverde** [comment]
Already solved :)
First I created an iso with "sudo eggs produce"
Then I copied ./local to two places:
To the "/etc/skel" directory next to .config.
And also to the directory created by eggs. Right at "/home/eggs/ovarium/filesystem.squashfs/home/user"
In the end I have recreated the iso with "sudo eggs produce"
And now it works perfect.

**Comment by pieroproietti** [comment]
Great, and compliments!
